lib/systems/platforms.nix

```diff
@@ -469,17 +468,27 @@ rec {
     '';
   };
 
-  selectBySystem = system: {
-      i486-linux = pc32;
-      i586-linux = pc32;
-      i686-linux = pc32;
-      x86_64-linux = pc64;
-      armv5tel-linux = sheevaplug;
-      armv6l-linux = raspberrypi;
-      armv7a-linux = armv7l-hf-multiplatform;
-      armv7l-linux = armv7l-hf-multiplatform;
-      aarch64-linux = aarch64-multiplatform;
-      mipsel-linux = fuloong2f_n32;
-      powerpc64le-linux = powernv;
-    }.${system} or pcBase;
+  # Deprecated 2020-11
+  selectBySystem = system:
+    lib.warn "`selectBySystem` is deprecated use `lib.systems.platforms.select`"
+    select (lib.systems.elaborate { inherit system; });
+
+  select = platform:
+    # x86
+    /**/ if platform.isx86_32 then pc32
```

Comment:
these should verify we are still looking at linux systems

Reply:
Actually the original didn’t check that for pcBase so maybe it shouldn’t matter. I guess defaulting yo pcBase if non linux is reasonable.

Reply:
Yeah I would rather decide what to do with that later. If some non-linux platform has some linux config stuff, it should be harmless.